polaris/evaluate/utils.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+import pandas as pd
+from typing import Optional
+
+from polaris.evaluate import BenchmarkResults, ResultsType
+from polaris.utils.types import PredictionsType
+from polaris.evaluate import Metric
+
+
+def is_multi_task_single_test_set(vals: PredictionsType, target_cols: list[str]):
+    """Check if the given values are for a multiple-task benchmark with a single
+    test set. This is inferred by comparing the target names with the keys of the
+    given data. If all keys in the given data match the target column names, we
+    assume they are target names (as opposed to test set names for a single-task,
+    multiple test set benchmark)."""
+    return all(k in target_cols for k in vals)
+
+
+def normalize_predictions_type(vals: PredictionsType, target_cols: list[str]):
+    if isinstance(vals, dict):
+        if is_multi_task_single_test_set(vals, target_cols):
+            return {"test": vals}
+        else:
+            return vals
+    elif vals is None:
+        return None
+    else:
+        return {"test": {target_cols[0]: vals}}
+
+
+def safe_mask(dict, test_label, target_label, mask):
+    if dict is None or dict[test_label] is None or dict[test_label][target_label] is None:
```

Comment:
**Question:** This feels off to me. Seems like it could raise a `KeyError`. Do you mean `dict.get(test_label)` rather than `dict[test_label]`? The difference being that if the `test_label` key is not in the dict, it will return `None` when using `get` and will throw an `KeyError` in the case of `[...]`.